apollo-router/src/axum_factory/tests.rs

```diff
@@ -2139,3 +2145,59 @@ async fn test_supergraph_and_health_check_same_port_different_listener() {
         error.to_string()
     );
 }
+
+#[tokio::test]
+async fn test_supergraph_timeout() {
+    let config = serde_json::json!({
+        "supergraph": {
+            "defer_support": false,
+        },
+        "traffic_shaping": {
+            "router": {
+                "timeout": "1ns"
+            }
+        },
+    });
+
+    let conf: Arc<Configuration> = Arc::new(serde_json::from_value(config).unwrap());
+
+    let schema = Schema::parse(
+        include_str!("..//testdata/minimal_supergraph.graphql"),
+        &conf,
+    )
+    .unwrap();
+
+    // we do the entire supergraph rebuilding instead of using `from_supergraph_mock_callback_and_configuration`
+    // because we need the plugins to apply on the supergraph
```

Comment:
Could you say more about what this means? The amount of duplicated boilerplate code feels unfortunate

Reply:
`from_supergraph_mock_callback_and_configuration` creates a mock supergraph without applying plugins on it, because it is meant as a foundation to test the router service. Here I want to test, at the router service response, the result of executing a plugin between the router service and supergraph service. That boiler plate could be extracted in another function if we see another case where we'd need it